syntax = "proto3";

package adlik.serving;
option cc_enable_arenas = true;

message ModelOperateRequest{
  string model_name = 1;
  string operate_name = 2;
  string path = 3;
  int64 model_version = 4;
}

message ModelOperateResponse{
  string status = 1; //SUCCESS, ERROR
  string error_message = 2;
  map<string, ModelInfo> models = 3;
}



message ModelInfo{
  message Version{
     int64 version_no = 1;
     string state = 2;
     string path = 3;
  }
  string model_path = 1;
  repeated Version versions = 2;
}
